### ISSUE TITLE
[leo_manager] Recover manager's RING in case storage's RING is correct in an automatic way

### ISSUE THREAD

**Issue by mocchira** [leo-project/leofs#1153]
### Related issues
- #1080

**Comment by yosukehara** [comment]
### Precondition

The current implementation of updating a `Cluster Member` table is done in LeoManager, then LeoManager distributes it to LeoStorage nodes *(this procedure is without exception)*. Therefore, since LeoManager's the records of `Cluster Member` table is master. If there is inconsistent records in the table of LeoStorage and LeoGateway, LeoFS delivers fixing inconsistent records under the premise.


### How To Validate

1. Validity check of a total number of RING's records
	* A sum total of vnodes which is configured at LeoStorage is equal with a total number of rows of RING table
	* Ref: [`leo_redundant_manager/include/leo_redundant_manager.hrl#L427`](https://github.com/leo-project/leo_redundant_manager/blob/1.9.64/include/leo_redundant_manager.hrl#L427)
2. Record validity of RING's records at stated periods
	*  Configures the execution intervals on `leo_storage.conf`
	*  Calculates the hash value of each RING table
		* Repeats until the last record: `Current.Record.HASH = MD5(Prev.Record.HASH + MD5(Current-Row-Record))`
3. Stores RING's records in memory *(Not store it into Mnesia, Erlang distributed database)*
	* LeoManager stores each RING table in memory as well as LeoStorage and LeoGateway
		* 1st because each RING table can be restored by `Cluster Member` table
		* 2nd because LeoManager can immediately find RING inconsistency by [1] and [2].


### How To Compare RING Between Manager(s) And Gateway, Storage Nodes

1. As the current implementation, LeoStorage nodes compare the hash value of each RING with a remote node, if it finds RING inconsistency, it sends a message to LeoManager to fix the inconsistency.
	* LeoManager investigates the RING of the node that sent the message, the node that seems to be inconsistent, and LeoManager to derive which node's RING is inconsistent.
2. LeoManager keeps correct RING by "[1] and [2] of How To Validate".
	* If LeoManager finds the wrong RING by itself, the RING is immediately restored by `Cluster Member` table. 

## In Conclusion

* As described above, we can reduce the implementation costs by utilizing the current implementation.
* We will replace its implementation with [leo_konsul](https://github.com/leo-project/leo_konsul) with LeoFS v1.5 or later.
